host/3.0/buster/amd64/java/java8/java8-build.Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM debian:buster
+
+ARG MAVEN_VERSION=3.8.4
+ARG USER_HOME_DIR="/root"
```

Comment:
USER_HOME_DIR as /root sounds strong permission. We should use the difference place like /home. 
@balag0 Which directory is preferable?

Reply:
I think /root is okay here as it is the home directory of the root user.
http://www.linfo.org/slash_root.html

using /home might cause conflicts since it is possible a network share gets mounted there based on user's config.

one thing you might have to verify is - how things work if the container is run as a non-root user. app service will run containers as non-root, so you can just test it there.

Reply:
@kaibocai Bala gave us a valuable suggestion.

Reply:
Thank you @balag0 @TsuyoshiUshio will test it on app service and update you.

Reply:
Hi @balag0 @TsuyoshiUshio , just test this build image using core tools on my local I see the java function can be created, build and start correctly without any issue. Please let me know if this test is sufficient. Thanks a lot!
![image](https://user-images.githubusercontent.com/89094811/149221141-5a9d2b82-7363-4ee7-8b08-b8dfddced345.png)

Reply:
Also by checking the maven docker file from MSFT https://github.com/microsoft/java/tree/master/docker/debian10, I see they are all use this /root folder. Thank you.

Reply:
great. oh so these images are meant to be used in dev machines only? and not on Azure?

Reply:
those image are used by core tools as here https://github.com/Azure/azure-functions-docker/blob/3f1e423536b13b9c318d7dfb34fb92c652cd3b1a/host/4/bullseye/amd64/java/java8/java8-core-tools.Dockerfile#L6, since we get rid of zulu jdk, those image are the one will have new msft jdks and these images only did 2 things, 1. install maven, 2. install jdk. Thank you.